### DISCUSSION TITLE
PETSC Error - after fresh system compilation

### DISCUSSION THREAD

**Discussion by Florian-Konrad** [idaholab/moose#24499] [Q&A Getting Started]
Hi there,

I have set up a new Ubuntu server and installed Moose (git commit 8664779e49 on 2023-05-16) and the app I use (GOLEM) without any problems just yesterday. When i want to start a test simulation which works perfectly on other machines I have but with older Moose versions (e.g. git commit d89c423467 on 2023-05-04) I get an error related to PETSC.

`PETSC ERROR: Incompatible vector local lengths parameter # 1 local size 76570 != parameter # 2 local size 76215`

See full terminal output below.
Any idea what is causing this?

Cheers
Florian

[petsc_libmesh_error.txt](https://github.com/idaholab/moose/files/11576507/petsc_libmesh_error.txt)

**Comment by lindsayad** [comment]
My suspicion is that the change affecting you is https://github.com/idaholab/moose/pull/24018. If you could tell me the test input you're running and remind me where the golem repo is (I believe it's on github) then I could look into this for you

**Reply by mcacace** [reply]
Thanks @lindsayad! The app that @Florian-Konrad is using is no longer on github but on an internal gitlab repo from my institution. I can add you to the members (would need an email associated to a valid git account) or Florian can rewrite the input syntax to match the old Golem version (on github) to replicate the problem.
Let me know and thanks again, 
Mauro

**Reply by lindsayad** [reply]
I would prefer to not have to leave github. If that becomes too complicated we can do the gitlab repo

**Comment by Florian-Konrad** [comment]
Thanks @mcacace and @lindsayad !

I just tested and I get the same error when using this Golem:
https://github.com/ajacquey/golem

The input I used to test is also here on github:
https://github.com/ajacquey/golem-examples/tree/main/THM_doublet

**Comment by ajacquey** [comment]
The golem GitHub repository (https://github.com/ajacquey/golem.git) has been updated (thanks @lindsayad for the support) and the input file in question runs fine with this repository.

**Reply by Florian-Konrad** [reply]
Hi @ajacquey did you test with the current Moose ? I just reinstalled everything clean and still get the same error. I'm using your Golem from main branch, is this right?

**Reply by mcacace** [reply]
@Florian-Konrad and @ajacquey trying to check what's the problem with the gitlab repo as well , but keep having error while rebuilding moose (undefined reference to `hit::Section::render). Once I manage to sort them out, I will try to see if I can be of any help here.

**Reply by mcacace** [reply]
Ok, I manage to install the current Moose. I'm on it ...

**Reply by mcacace** [reply]
So, @ajacquey I think the test passed because running on single processor. The problem only occurs if running on multiple processors. 
This said, I managed to have it run while moving the declaration of the displacements vector out from the GlobalParams block.  I remembered we had a similar issue in the past, possibly being related to a different partitioning between the displaced and undisplaced mesh. @lindsayad  can it be a duplicated of [https://github.com/idaholab/moose/issues/12799](url)?

**Reply by lindsayad** [reply]
@mcacace that link gives me a 404

**Reply by mcacace** [reply]
Sorry wrong link (https://groups.google.com/g/moose-users/c/RI-2ZEiKzms/m/1ekZbofMDgAJ)

**Reply by lindsayad** [reply]
It would be helpful if you could provide a smaller mesh for this problem, but maybe that's not possible.

The link is a good one. I expect it very well could be different partitions between undisplaced and displaced meshes

**Reply by lindsayad** [reply]
Ok https://github.com/idaholab/moose/pull/24550 seems to fix the issue for me

**Reply by mcacace** [reply]
Thanks! Let me know if you need any assistance from my side.

**Reply by lindsayad** [reply]
Maybe just test it out and let me know if that PR works for you?

**Comment by mcacace** [comment]
Will do. Thanks again
